src/connector/src/sink/snowflake_redshift/redshift.rs

```diff
@@ -460,97 +397,181 @@ impl RedshiftSinkCommitter {
-        let mut interval_timer = interval(Duration::from_secs(schedule_seconds)); // 1 hour = 3600 seconds
-        interval_timer.set_missed_tick_behavior(MissedTickBehavior::Skip);
-        let sql = build_create_merge_into_task_sql(
-            schema_name,
-            cdc_table_name,
-            target_table_name,
-            &pk_column_names,
-            &all_column_names,
-        );
+        let mut copy_timer = interval(Duration::from_secs(write_intermediate_interval_seconds));
+        copy_timer.set_missed_tick_behavior(MissedTickBehavior::Skip);
+        let mut merge_timer = interval(Duration::from_secs(writer_target_interval_seconds));
+        merge_timer.set_missed_tick_behavior(MissedTickBehavior::Skip);
+
         loop {
             tokio::select! {
-                // Check for shutdown signal
-                _ = shutdown_receiver.recv() => {
-                    tracing::info!("Periodic query task received shutdown signal, stopping");
-                    break;
-                }
-                // Execute periodic query
-                _ = interval_timer.tick() => {
-
-                    match client.execute_sql_sync(sql.clone()).await {
-                        Ok(_) => {
-                            tracing::info!("Periodic query executed successfully for table: {}", target_table_name);
-                        }
-                        Err(e) => {
-                            tracing::warn!("Failed to execute periodic query for table {}: {}", target_table_name, e.as_report());
-                        }
+                _ = shutdown_receiver.recv() => break,
+                _ = merge_timer.tick(), if merge_into_sql.is_some() => {
+                    if let Some(sql) = &merge_into_sql && let Err(e) = client.execute_sql_sync(sql.clone()).await {
+                            tracing::warn!("Failed to execute periodic query for table {}: {}", config.table, e);
+
+                    }
+                },
+                _ = copy_timer.tick(), if need_copy_into => {
+                    if let Err(e) = async {
+                        let s3_inner = config.s3_inner.as_ref().ok_or_else(|| {
+                            SinkError::Config(anyhow!("S3 configuration is required for redshift s3 sink"))
+                        })?;
+                        let manifest_dir = manifest_dir.as_ref().ok_or_else(|| {
+                            SinkError::Config(anyhow!("Manifest directory is not initialized"))
+                        })?;
+                        let manifest_dir_str = {
+                            let mut manifest_dir_guard = manifest_dir.write().await;
+                            let manifest_dir_str = manifest_dir_guard.clone();
+                            *manifest_dir_guard = build_new_manifest_dir();
+                            manifest_dir_str
+                        };
+                        Self::copy_into_from_s3_to_redshift(&client, &config,s3_inner, false, &manifest_dir_str).await?;
```

Comment:
`is_append_only` is always false when using `copy_into_from_s3_to_redshift`. is it expected?